From 28924ab4c414b4588f45beae7190e44680d6701b Mon Sep 17 00:00:00 2001
From: yuanxia <yuanxia2073@phytium.com.cn>
Date: Mon, 1 Jul 2024 10:01:35 +0800
Subject: [PATCH 099/360] dt-bindings: phytnet_led: Add bindings for Phytium
 MAC led

This patch document the DT bindings for gmac led controller.

Signed-off-by: yuanxia <yuanxia2073@phytium.com.cn>
Signed-off-by: Li Wencheng <liwencheng@phytium.com.cn>
Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>
---
 .../devicetree/bindings/leds/phytnet_led.yaml | 46 +++++++++++++++++++
 1 file changed, 46 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/leds/phytnet_led.yaml

diff --git a/Documentation/devicetree/bindings/leds/phytnet_led.yaml b/Documentation/devicetree/bindings/leds/phytnet_led.yaml
new file mode 100644
index 000000000000..24efebc9ba30
--- /dev/null
+++ b/Documentation/devicetree/bindings/leds/phytnet_led.yaml
@@ -0,0 +1,46 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/net/phytnet_led.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium mac led controller
+
+maintainers:
+  - LongShixiang <longshixiang1718@phytium.com.cn>
+
+description:
+  This modules is used to control mac led.
+
+properties:
+  compatible:
+    const: phytium,net_led
+  net_dev:
+    maxItems: 1
+    description: Phandler of specified Net device
+  led-gpios:
+    minItems: 1
+    maxItems: 2
+    description: |-
+      the gpios used for led control based on net_dev condition.
+      One represents LINK condition, another represents ACT condition.
+
+required:
+  - compatible
+  - net_dev
+  - led-gpios
+
+examples:
+  - |
+    gpiochip0: gpop_controller{
+            ...
+    }
+    eth0: ethernet{
+            ...
+    }
+    phytium_net_led0 {
+      compatible = "phytium,net_led";
+      net_dev = <&eth1>;
+      led-gpios = <&gpiochip0 9 GPIO_ACTIVE_HIGH>,  /* link */
+                  <&gpiochip0 11 GPIO_ACTIVE_HIGH>; /* act */
+    };
-- 
2.20.1

